### DISCUSSION TITLE
Help me set comments to italics in Lazy + catppuccin

### DISCUSSION THREAD

**Discussion by jjaimealeman** [LazyVim/LazyVim#2027] [Q&A]
I've tried all these.

I just can't get `styles.comments` to format as italic 🤔

```lua
return {
  { "catppuccin/nvim" },
  lazy = true,
  name = "catppuccin",
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "catppuccin",
      flavour = "mocha",
      background = {
        light = "latte",
        dark = "mocha",
      },
      styles = {
        comments = { "italic" },
      },
    },
  },
}
```

```lua
return {
  {
    "catppuccin/nvim",
    lazy = true,
    name = "catppuccin",
    -- you can do it like this with a config function
    config = function()
      require("catppuccin").setup {
        -- configurations
      }
    end,
      colorscheme = "catppuccin",
      flavour = "mocha",
      styles = {
        -- comments = { "italic", "underdotted" },
        comments = { "italic" },
      },
  },
  {
    "LazyVim/LazyVim",
  },
}
```

```lua
return {
  {
    "catppuccin/nvim",
    lazy = true,
    name = "catppuccin",
    -- you can do it like this with a config function
    config = function()
      require("catppuccin").setup({
        -- configurations
        colorscheme = "catppuccin",
        flavour = "mocha",
        styles = {
          comments = { "italic", "underdotted" },
          -- comments = { "italic" },
        },
      })
    end,
  },
  {
    "LazyVim/LazyVim",
  },
}
```

**Comment by dpetka2001** [comment]
That depends on the font you use in your terminal. Also the correct config for doing what you want would be something like 
```lua
return {
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "catppuccin",
    },
  },

  -- test comment
  {
    "catppuccin",
    opts = {
      flavour = "mocha",
      styles = {
        comments = { "italic", "underline" },
      },
    },
  },
}
```
As long as you configure your terminal with fonts that support italic the above snippet should do what you want.

**Comment by dimas-cyriaco** [comment]
You can use this snippet to test if your terminal is rendering the fonts ok:

```sh
echo -e "\e[1mbold\e[0m"
echo -e "\e[3mitalic\e[0m"
echo -e "\e[3m\e[1mbold italic\e[0m"
echo -e "\e[4munderline\e[0m"
echo -e "\e[9mstrikethrough\e[0m"
```

**Reply by jjaimealeman** [reply]
_AAH, i see!_

Thanks for this!

Fira Code won't display as italic. It looks like I'm headed to download some new fonts then.

https://www.nerdfonts.com/ 👍

---

Any recommendations for a good coding font, that looks good, has italics, AND optionally, has nice looking ligatures?

**Reply by dimas-cyriaco** [reply]
I use`JetBrainsMono Nerd Font`. Very good font.

**Reply by cerebrl** [reply]
I think IBM Plex is one of the greatest type families out there: https://www.ibm.com/plex (also [on Google Fonts here](https://fonts.google.com/?query=ibm+plex)). I cannot recommend this typeface enough as it's the perfect blend of utility, versatility and beauty (yes, I'm a type snob). Of course, for code, you'd use the Mono variation of the typeface family. I use the other non-monospace fonts for Obsidian and other non-code writing.

I have not tried to use coding ligatures with it, so keep that in mind, but for all other purposes, it shines (IMO). Here's a screenshot of it from my Neovim:

![CleanShot 2023-11-16 at 18 17 25](https://github.com/LazyVim/LazyVim/assets/3070458/d4d18b33-e4b4-4d61-8053-81c5d6c765bf)

**Reply by jjaimealeman** [reply]
WELL, turns out I already had IBMPlex installed 🤣

`kitty + list-fonts --psnames | grep IBMPlex` just had to add it real quick to kitty.

And my `catpuccin.lua` was properly set.

Here's what it all looks like, properly set up!
![image](https://github.com/LazyVim/LazyVim/assets/1428292/592e3fee-27c1-4f14-a717-9595675a9e78)


---

For anyone wanting to set up italics for a sexy LazyVim experience:

`catpuccin.lua`
```lua
return {
  {
    "catppuccin/nvim",
    lazy = true,
    name = "catppuccin",
    config = function()
      require("catppuccin").setup({
        -- configurations
        colorscheme = "catppuccin",
        flavour = "mocha",
        styles = {
          comments = { "italic" },
        },
      })
    end,
    integrations = {
      aerial = true,
      alpha = true,
      cmp = true,
      dashboard = true,
      flash = true,
      gitsigns = true,
      headlines = true,
      illuminate = true,
      indent_blankline = { enabled = true },
      leap = true,
      lsp_trouble = true,
      mason = true,
      markdown = true,
      mini = true,
      native_lsp = {
        enabled = true,
        underlines = {
          errors = { "undercurl" },
          hints = { "undercurl" },
          warnings = { "undercurl" },
          information = { "undercurl" },
        },
      },
      navic = { enabled = true, custom_bg = "lualine" },
      neotest = true,
      neotree = true,
      noice = true,
      notify = true,
      semantic_tokens = true,
      telescope = true,
      treesitter = true,
      treesitter_context = true,
      which_key = true,
    },
  },
  {
    "LazyVim/LazyVim",
  },
}
```
